src/main/java/com/ironcorelabs/tenantsecurity/kms/v1/TenantSecurityKMSClient.java

```diff
@@ -376,6 +443,58 @@ public static boolean isCiphertext(byte[] bytes) {
         return CompletableFutures.sequence(encryptOps);
     }
 
+    /**
+     * Encrypt a map of documents from the ID of the document to the list of fields
+     * to encrypt. Makes a call out to the Tenant Security Proxy to generate a
+     * collection of new DEK/EDEK pairs for each document ID provided. This function
+     * supports partial failure so it returns two Maps, one of document ID to
+     * successfully encrypted document and one of document ID to a
+     * TenantSecurityKMSException.
+     *
+     * @param plaintextDocuments Map of document ID to map of fields to encrypt.
+     * @param metadata           Metadata about all of the documents being encrypted
+     * @return Collection of successes and failures that occured during operation.
+     *         The keys of each map returned will be the same keys provided in the
+     *         original plaintextDocuments map.
+     */
+    public CompletableFuture<BatchResult<EncryptedDocument>> encryptBatchBetter(
```

Comment:
My vote is still to just name this `encryptBatch` and throw `@Deprecated` on the old ones. If we were on Java9 `    @Deprecated(since = "2", forRemoval = true)`.

Reply:
Just to clarify, you mean create a second overload for these methods and leave the old ones with the same type signature?